t/05-scheduler-full.t

```diff
@@ -136,146 +136,6 @@ subtest 'Scheduler worker job allocation' => sub {
-    dead_workers($schema);
-
-    # simulate a worker in idle state that rejects all jobs assigned to it
-    @workers = rejective_worker(@$worker_settings, 3, 'rejection reason');
-    wait_for_worker($schema, 5);
-
-    note 'waiting for job to be assigned and set back to re-scheduled';
-    # the loop is needed as the scheduler sometimes needs a second
-    # cycle before the worker is seen as unusable
-    for (1 .. 2) {
-        $allocated = scheduler_step();
-        last if $allocated && @$allocated >= 1;
-        note "scheduler could not yet assign to rejective worker, try: $_";    # uncoverable statement
-    }
-    is @$allocated, 1, 'one job allocated'
-      and is @{$allocated}[0]->{job},    99982, 'right job allocated'
-      and is @{$allocated}[0]->{worker}, 5,     'job allocated to expected worker';
-    my $job_assigned  = 0;
-    my $job_scheduled = 0;
-    for (0 .. 100) {
-        my $job_state = $jobs->find(99982)->state;
-        if ($job_state eq OpenQA::Jobs::Constants::ASSIGNED) {
-            note 'job is assigned' unless $job_assigned;
-            $job_assigned = 1;
-        }
-        elsif ($job_state eq OpenQA::Jobs::Constants::SCHEDULED) {
-            $job_scheduled = 1;
-            last;
-        }
-        sleep .2;
-    }
-    ok $job_scheduled, 'assigned job set back to scheduled if worker reports back again but has abandoned the job';
-    stop_workers;
-    dead_workers($schema);
-
-    # start an unstable worker; it will register itself but ignore any job assignment (also not explicitely reject
-    # assignments)
-    @workers = unstable_worker(@$worker_settings, 3, -1);
-    wait_for_worker($schema, 5);
-    for (1 .. 2) {
-        $allocated = scheduler_step();
-        last if $allocated && @$allocated >= 1;
-        note "scheduler could not yet assign to broken worker, try: $_";    # uncoverable statement
-    }
-    is @$allocated, 1, 'one job allocated'
-      and is @{$allocated}[0]->{job},    99982, 'right job allocated'
-      and is @{$allocated}[0]->{worker}, 5,     'job allocated to expected worker';
-
-    # kill the worker but assume the job has been actually started and is running
-    stop_workers;
-    $jobs->find(99982)->update({state => OpenQA::Jobs::Constants::RUNNING});
-
-    @workers = unstable_worker(@$worker_settings, 3, -1);
-    wait_for_worker($schema, 5);
-
-    note 'waiting for job to be incompleted';
-    for (0 .. 100) {
-        last if $jobs->find(99982)->state eq OpenQA::Jobs::Constants::DONE;
-        sleep .2;
-    }
-
-    my $job = $jobs->find(99982);
-    is $job->state, OpenQA::Jobs::Constants::DONE,
-      'running job set to done if its worker re-connects claiming not to work on it anymore';
-    is $job->result, OpenQA::Jobs::Constants::INCOMPLETE,
-      'running job incompleted if its worker re-connects claiming not to work on it anymore';
-    like $job->reason, qr/abandoned: associated worker .+:\d+ re-connected but abandoned the job/, 'reason is set';
```

Comment:
Not sure whether this is already covered (and also actually checked!) somewhere else.

Reply:
I was hoping to see coverage results telling me that.